document/js-api/index.bs

```diff
@@ -924,7 +924,7 @@ Note: Exported Functions do not have a \[[Construct]] method and thus it is not
             1. If |method| is undefined, [=throw=] a {{TypeError}}.
             1. Let |values| be ? [=IterableToList=](|ret|, |method|).
             1. Let |wasmValues| be a new, empty [=list=].
-            1. Assert: |values| and |results| have the same length.
+            1. If |values| and |results| do not have the same length, trigger a WebAssembly trap, then throw a {{TypeError}} to the enclosing JavaScript.
```

Comment:
Let's not add the trap wording here; it should be covered when all this moves into the ["run a host function"](https://webassembly.github.io/spec/js-api/#create-a-host-function) that exists in the base spec.